client-sdk/ts-web/rt/src/callformat.ts

```diff
@@ -1,4 +1,93 @@
+import * as oasis from '@oasisprotocol/client';
+
+import * as transaction from './transaction';
+import * as mrae from './mrae';
+import * as types from './types';
+
 /**
  * Call data key pair ID domain separation context base.
  */
 export const CALL_DATA_KEY_PAIR_ID_CONTEXT_BASE = 'oasis-runtime-sdk/private: tx';
+
+/**
+ *  EncodeConfig is call encoding configuration.
+ *  golang: oasis-sdk/client-sdk/go/callformat/callformat.go
+ *  rust:
+ */
+export interface EncodeConfig {
+    /**
+     * publicKey is an optional runtime's call data public key to use for encrypted call formats.
+     */
+    publicKey?: types.KeyManagerSignedPublicKey;
+}
+
+export interface MetaEncryptedX25519DeoxysII {
+    sk: Uint8Array;
+    pk: Uint8Array;
+}
+
+/**
+ * encodeCall encodes a call based on its configured call format.
+ * It returns the encoded call and any metadata needed to successfully decode the result.
+ */
+export async function encodeCall(
+    nonce: Uint8Array,
+    sk: Uint8Array,
+    pk: Uint8Array,
```

Comment:
these are pretty specialized for the deoxysii encryption. I see you've added these in response to nhynes's request to make a keypair reusable. I would like for us still to have a variant that internally generates the nonce and keys though, so that we have parity with the go client sdk's functionality.

at this point, the function signature so specialized that maybe it would be better to have separate encoding+decoding functions for each supported callformat. do you have any use cases in mind where the callsite doesn't know which callformat is chosen and would prefer to use a general encodeCall+decodeResult function?

Reply:
added, thanks